pkg/api/telemetry.go

```diff
@@ -143,10 +143,22 @@ func NewTelemetryEventFromKubernetesJobEvent(event *pkg.JobEvent, logs string, p
 		Timestamp: strconv.FormatInt(eventTime.Unix(), 10),
 	}
 
-	if env := pkg.NewCronitorConfigParser(cronjob).GetEnvironment(); env != "" {
+	cronitorConfigParser := pkg.NewCronitorConfigParser(cronjob)
+
+	if env := cronitorConfigParser.GetEnvironment(); env != "" {
 		telemetryEvent.Env = env
 	}
 
+	// Fixes issue https://github.com/cronitorio/cronitor-kubernetes/issues/47
+	// When auto-complete is disabled via 'k8s.cronitor.io/auto-complete: "false"' annotation,
+	// we send "Completed" events as "logs" type instead of "complete" to prevent cronitor.io
+	// from auto-completing the monitor. This supports async workflows where the actual task
+	// completion happens outside the Kubernetes job.
+	if ok, _ := cronitorConfigParser.EnabledAutoComplete(); !ok && telemetryEvent.Event == Complete {
```

Comment:
once this is renamed we'll have to reverse the boolean logic here again...sorry!